### ISSUE TITLE
TomcatService does not supports AsyncServlet.

### ISSUE THREAD

**Issue by blmarket** [line/armeria#116]
I'd like to use async feature like `SseEmitter` but armeria does not supports these features yet.

**Comment by trustin** [comment]
Yeah, I tried to get this work, but could not figure out how. Care to dive deeeeep?

**Comment by blmarket** [comment]
It's too deep, so taking deeeeeep breath first.

our use cases are experimental yet, so will wait until it's going serious consideration. currently we're opening both tomcat & armeria ports to workaround.

**Comment by imasahiro** [comment]
I've met simler situation when implementing async API with spring-boot (using `DeferredResult`) + `TomcatService` (and not solved yet).
Here is minimum sample code for our case.
https://github.com/imasahiro/armeria-issue/blob/issue116/src/main/java/com/linecorp/armeria/sample/SampleController.java#L22

**Comment by alex-lx** [comment]
Met the same situation when using `DeferredResult`

**Comment by minwoox** [comment]
Spring GraphQL also does not work when it's used with `armeria-spring-bootx-autoconfigure` module because AsyncServlet isn't supported.
However, Spring GraphQL works if it's used with the webflux integration module.

**Comment by nao0811ta** [comment]
Still, it's not available? How will Kotlin behavior? that uses `suspend` as an async

**Comment by minwoox** [comment]
No, it still doesn't support AsyncServlet. @nao0811ta Are you using it with Kotlin?

**Comment by nao0811ta** [comment]
Sorry. it wasn't Armeria. never mind
